src/mmw/apps/modeling/calcs.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import absolute_import
+
+from collections import namedtuple
+
+from django.db import connection
+
+from django.contrib.gis.geos import GEOSGeometry
+
+
+def animal_population(geojson):
+    """
+    Given a geojson shape, calculate the area-weighted county animal
+    population. Returns a dictionary to append to the outgoing JSON.
+    """
+    geom = GEOSGeometry(geojson, srid=4326)
+
+    sql = '''
+        WITH clipped_counties AS (
+            SELECT ST_Intersection(geom,
+                                   ST_SetSRID(ST_GeomFromText(%s),
+                                              4326)) AS geom_clipped,
+                   ms_county_animals.*
+            FROM ms_county_animals
+            WHERE ST_Intersects(geom,
+                                ST_SetSRID(ST_GeomFromText(%s),
+                                           4326))
+        ), clipped_counties_with_area AS (
+            SELECT ST_Area(geom_clipped) /
+                   ST_Area(ST_SetSRID(ST_GeomFromText(%s),
+                                      4326)) AS clip_percent,
+                   clipped_counties.*
+            FROM clipped_counties
+        )
+        SELECT SUM(beef_ha * clip_percent) as beef_ha,
+               SUM(dairy_ha * clip_percent) as dairy_ha,
+               SUM(broiler_ha * clip_percent) as broiler_ha,
+               SUM(goat_ha * clip_percent) as goat_ha,
+               SUM(hog_ha * clip_percent) as hog_ha,
+               SUM(horse_ha * clip_percent) as horse_ha,
+               SUM(layer_ha * clip_percent) as layer_ha,
+               SUM(sheep_ha * clip_percent) as sheep_ha,
+               SUM(turkey_ha * clip_percent) as turkey_ha,
+               SUM(totalaeu * clip_percent) as totalaeu,
+               SUM(grazingaeu * clip_percent) as grazingaeu,
+               SUM(totaeu_ha * clip_percent) as totaeu_ha,
+               SUM(grzaeu_ha * clip_percent) as grzaeu_ha,
+               SUM(area * clip_percent) as area,
+               SUM(totalha * clip_percent) as totalha,
+               SUM(ag_ha * clip_percent) as ag_ha,
+               SUM(ctyareaha * clip_percent) as ctyareaha,
+               SUM(ag_acres * clip_percent) as ag_acres
+        FROM clipped_counties_with_area;
+        '''
+
+    with connection.cursor() as cursor:
+        cursor.execute(sql, [geom.wkt, geom.wkt, geom.wkt])
+
+        animal_pop = namedtuple('AnimalPopulation',
+                                [col[0] for col in cursor.description])
+
+        animal_data = {}
+
+        for k, v in animal_pop(*(cursor.fetchone()))._asdict().iteritems():
+            animal_data[k.decode('utf-8')] = v
```

Comment:
Instead of transforming the query results into a named tuple and then into a dictionary, consider transforming directly to a dictionary as done here: https://github.com/WikiWatershed/model-my-watershed/blob/9cc43eff1dc8fdebf7a07c7ee6c76ecaf8dd2ace/src/mmw/apps/modeling/mapshed/calcs.py#L183-L186